src/utarray.h

```diff
@@ -58,7 +58,7 @@ typedef struct {
 } UT_icd;
 
 typedef struct {
-    unsigned i,n;/* i: index of next available slot, n: num slots */
+    size_t i,n;/* i: index of next available slot, n: num slots */
```

Comment:
I'd prefer not to change the layout of `struct UT_array` if we can help it.

I infer that the rest of the changes in this patch are cascading from this one: since you changed the type of `i`, the expression `_ut_i < (a)->i` started warning, so you also had to change the type of `_ut_i`, and so on.

Suppose you leave this line alone, and approach the problem from the other end: where are the individual expressions that trigger the MSVC diagnostic in master, and can we just add casts or other local changes to silence the warnings down there?

Reply:
Actually changing the struct was the **last** thing I did. I tried your suggestion, ie patching individual lines with casts, to no avail.
So in the end I gave up and did a systematic search replace of `unsigned` / `size_t` and that fixed it.
I do agree that changing the struct just doesn't seem right, so I'll have another go at it.